src/OpenTelemetry/BaseProcessor.cs

```diff
@@ -17,7 +17,9 @@ public abstract class BaseProcessor<T> : IDisposable
     /// <summary>
     /// Initializes a new instance of the <see cref="BaseProcessor{T}"/> class.
     /// </summary>
+#pragma warning disable CA1012 // Abstract types should not have public constructors
```

Comment:
Is it better to change it to `protected`?
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1012

> To fix a violation of this rule, either make the constructor protected or don't declare the type as abstract.

Reply:
> Is it better to change it to `protected`? https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1012
> 
> > To fix a violation of this rule, either make the constructor protected or don't declare the type as abstract.

I have tried this. It is failing on backward compatibility verification. I decided to do not introduced any binary breaking changes.

Reply:
> If not now, should we change it later? Maybe we should create an issue, or things like this would easily get lost.

If we decided to go with major release, we should try to remove all disabled warnings. It is not related only to this. As we discussed ~month ago it is unlikely scenario. I doubt that it is worth to create now anything like this.